src/TorchSharp/NN/Module.cs

```diff
@@ -461,13 +461,19 @@ public virtual IEnumerable<(string name, Module module)> named_modules()
                 /// <returns>(string, Parameter) – Tuple containing the name and parameter</returns>
                 public virtual IEnumerable<(string name, Modules.Parameter parameter)> named_parameters(bool recurse = true)
                 {
+                    var seen = new HashSet<IntPtr>();
+
                     foreach (var nsm in _internal_params) {
+                        if (seen.Contains(nsm.Item2.Handle)) continue;
+                        seen.Add(nsm.Item2.Handle);
                         yield return nsm;
                     }
 
                     if (recurse) {
                         foreach (var (name, sm) in _internal_submodules) {
                             foreach (var (n, p) in sm.named_parameters(true)) {
+                                if (seen.Contains(p.Handle)) continue;
+                                seen.Add(p.Handle);
                                 yield return ($"{name}.{n}", p);
```

Comment:
seen should be passed recursively

Reply:
I guess that would be more efficient, but it shouldn't affect the result. I'll have to create a non-public member for it.

Reply:
You are right, I did not notice it was not actually a counting function, but a parameter enumerator.

Reply:
It's still a good idea to pass it down for efficiency.